src/datetime.rs

```diff
@@ -209,22 +209,22 @@ impl<U: Unit> fmt::Debug for Timedelta<U> {
 
 struct TypeDescriptors {
     npy_type: NPY_TYPES,
-    #[allow(clippy::type_complexity)]
-    dtypes: GILProtected<RefCell<Option<FxHashMap<NPY_DATETIMEUNIT, Py<PyArrayDescr>>>>>,
+    dtypes: Mutex<Option<FxHashMap<NPY_DATETIMEUNIT, Py<PyArrayDescr>>>>,
 }
 
 impl TypeDescriptors {
     /// `npy_type` must be either `NPY_DATETIME` or `NPY_TIMEDELTA`.
     const unsafe fn new(npy_type: NPY_TYPES) -> Self {
         Self {
             npy_type,
-            dtypes: GILProtected::new(RefCell::new(None)),
+            dtypes: Mutex::new(None),
         }
     }
 
     #[allow(clippy::wrong_self_convention)]
     fn from_unit<'py>(&self, py: Python<'py>, unit: NPY_DATETIMEUNIT) -> Bound<'py, PyArrayDescr> {
-        let mut dtypes = self.dtypes.get(py).borrow_mut();
+        // FIXME probably a deadlock risk here due to the GIL? Might need MutexExt trait in PyO3
```

Comment:
Missed this on my first pass, this should use the manual deadlock avoidance used in `datetime.rs`.